libminifi/include/core/ContentRepository.h

```diff
@@ -28,20 +28,18 @@
 #include "core/Connectable.h"
 #include "ContentSession.h"
 #include "utils/GeneralUtils.h"
+#include "core/Repository.h"
 
 namespace org::apache::nifi::minifi::core {
 
 /**
  * Content repository definition that extends StreamManager.
  */
-class ContentRepository : public StreamManager<minifi::ResourceClaim>, public utils::EnableSharedFromThis<ContentRepository> {
+class ContentRepository : public virtual core::Repository, public StreamManager<minifi::ResourceClaim>, public utils::EnableSharedFromThis<ContentRepository> {
```

Comment:
Why is the inheritance from core::Repository needed for ContentRepository? There is a discussion about a common base class introduced in #1490 (https://github.com/apache/nifi-minifi-cpp/pull/1490#discussion_r1093458349). A suggested change is that core::Repository should be renamed as it represents the a key-value based repository while content repo has a stream based interface so it should not be a common base class. This will probably conflict with #1485 maybe we should try rebasing this change on that.

Reply:
I am fine with waiting until the mentioned PRs are merged

Reply:
in light of changes since and future plans, rewrote this PR to not use inheritance